### PR TITLE
Fix stale kvstore lock timeout behavior

### PR DESCRIPTION
**Pull request by tgraf**
See commits for details

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6699)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: pkg/kvstore/lock.go [hunk 1]
```diff
@@ -58,9 +65,9 @@ func (pl *pathLocks) lock(path string) {
 			return
 		}
 
-		if time.Since(started) > lockTimeout {
+		if time.Since(started) > staleLockTimeout {
 			log.WithField("path", path).Warning("WARNING: Timeout while waiting for lock, ignoring lock")
-			pl.lockPaths[path] = 1
+			pl.lockPaths[path] = 0
 			pl.mutex.Unlock()
```
Comment (aanm) [inline pkg/kvstore/lock.go:71]: @tgraf how will the caller distinguish that the lock was really gotten or not?

**Review by aanm — CHANGES_REQUESTED** [review]

File: pkg/kvstore/lock.go (hunk 1 above)
Comment (tgraf) [reply pkg/kvstore/lock.go:71]: I've changed it to restart the acquisition cycle. The downside of this is a theoretical chance of permanent starvation but handles the case when multiple local consumers are waiting on the same key.

**Comment by tgraf** [comment]
test-me-please

File: pkg/kvstore/lock.go (hunk 1 above)
Comment (aanm) [reply pkg/kvstore/lock.go:71]: :+1: please change the "ignoring lock" part in the comment, otherwise looks good

**Review by aanm — APPROVED** [review]

**Pull request merged**